sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -663,13 +680,8 @@ private static SerializerAdapter createDefaultSerializer() {
      * @return the default HttpPipeline
      */
     private static HttpPipeline createDefaultPipeline() {
-        List<HttpPipelinePolicy> policies = new ArrayList<>();
-        policies.add(new UserAgentPolicy());
-        policies.add(new RetryPolicy());
-        policies.add(new CookiePolicy());
-
         return new HttpPipelineBuilder()
-            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .policies(new UserAgentPolicy(), new RetryPolicy(), new CookiePolicy())
```

Comment:
these could be static singletons if we want to be even faster.

Reply:
Good idea, don't know if this is ever called, but we need to keep it since it's public API (unfortunately).